### PR TITLE
Better logs for parts of the codebase that might be expensive V2

### PR DESCRIPTION
**Pull request by shams-ahmed**
**Context**
I looked at areas of the codebase that lacked debugging logs and added timestamps to the areas where we might need to review them in the future.

Also, added quite a few logs that will need to be removed once we get a better feel and assess the performance.

**Changes**
- lots of logs
- added timestamp
- logs of how many targets are being processed.

### CODE REVIEW

**Comment by lpusok** [comment]
Would it be possible to use debug logs instead of fmt.Printf()? That for one would use the logger interface for the output, which we already use most places. An other reason to do that is to keep the customer-facing logs readable.

**Comment by shams-ahmed** [comment]
@lpusok Made a few adjustments moving some logs from `.info/.print` to `.debug`. It's a hard tradeoff but giving we know so little about performance issues I had to make some logs more verbose and noisy. Hopefully we can remove some of the logs soon as we learn some of the root issues.

File: autocodesign/projectmanager/projecthelper.go [hunk 1]
```diff
@@ -440,6 +444,8 @@ func mainTargetOfScheme(proj xcodeproj.XcodeProj, scheme xcscheme.Scheme) (xcode
 // findBuiltProject returns the Xcode project which will be built for the provided scheme, plus the scheme.
 // The scheme is returned as it could be found under the .xcworkspace, and opening based on name from the XcodeProj would fail.
 func findBuiltProject(pth, schemeName string) (xcodeproj.XcodeProj, xcscheme.Scheme, error) {
+	log.TInfof("Location built project for xcode project: %s, scheme: %s", pth, schemeName)
```
[the remark this answers is not in the document]
Comment (shams-ahmed) [reply autocodesign/projectmanager/projecthelper.go:447]: updated

File: exportoptionsgenerator/exportoptionsgenerator.go [hunk 2]
```diff
@@ -86,6 +89,8 @@ func (g ExportOptionsGenerator) GenerateApplicationExportOptions(exportMethod ex
 		}
 	}
 
+	g.logger.TDebugf("Generated application export options entitlements for: %s", exportMethod)
```
[the remark this answers is not in the document]
Comment (shams-ahmed) [reply exportoptionsgenerator/exportoptionsgenerator.go]: updated

File: xcpretty/xcpretty.go [hunk 3]
```diff
@@ -84,11 +84,11 @@ func (c CommandModel) Run() (string, error) {
 	// Always close xcpretty outputs
 	defer func() {
 		if err := pipeWriter.Close(); err != nil {
-			log.Warnf("Failed to close xcodebuild-xcpretty pipe, error: %s", err)
+			fmt.Printf("Failed to close xcodebuild-xcpretty pipe, error: %s", err)
```
[the remark this answers is not in the document]
Comment (shams-ahmed) [reply xcpretty/xcpretty.go]: Updated the PR.. changed it because of dependencies issues.

**Review by godrei — APPROVED** [review]

**Pull request merged**